source/MaterialXCore/Document.cpp

```diff
@@ -1094,8 +1101,36 @@ void Document::upgradeVersion(bool applyFutureUpdates)
                     node->setName(newNodeName);
                 }
             }
+        }   
+
+        // Convert all parameters to be inputs. If needed set them to be "uniform".
+        for (ElementPtr e : traverseTree())
+        {
+            InterfaceElementPtr elem = e->asA<InterfaceElement>();
+            if (!elem)
+            {
+                continue;
+            }
+            vector<ElementPtr> children = elem->getChildren();
+            for (ElementPtr child : children)
+            {
+                const string& childName = child->getName();
+                if (child->isA<Parameter>())
```

Comment:
Would the updateChildSubclass helper function work here?

Reply:
Ah yes. This utility is much better to use.